src/main/java/com/rabbitmq/client/impl/recovery/DefaultRetryHandler.java

```diff
@@ -122,11 +122,11 @@ protected RetryResult doRetry(BiPredicate<RecordedEntity, Exception> condition,
                 throw exception;
             }
         }
-        throw context.exception();
+        throw exception;
```

Comment:
assuming we want the last exception thrown here rather than the 1st? If not i can change it back.

Reply:
Makes sense.